src/ClassDesignChallengeConstructors/Professor.java

```diff
@@ -0,0 +1,20 @@
+package ClassDesignChallengeConstructors;
+
+public class Professor {
+
+    private String firstName;
+    private String lastName;
+    private String cnp;
+    private String addressCity;
+    private String addressStreet;
+    private int addressApartmentNr;
```

Comment:
What if the professor has a house, not an apartment? We can find a generic naming for this property? Maybe `addressNumber`?
It might be a good idea to have an Address class (with country, city, street, and number properties) that can be used in Professor and Student classes?